app/models/purchase.rb

```diff
@@ -3389,7 +3389,15 @@ def calculate_installment_payment_price_cents(total_price_cents)
       return unless is_installment_payment
 
       nth_installment = subscription&.purchases&.successful&.count || 0
-      installment_payments = fetch_installment_plan.calculate_installment_payment_price_cents(total_price_cents)
+      payment_option = subscription&.last_payment_option
+
+      # Prefer snapshot to lock customer's original terms, fallback to live plan for backwards compatibility
```

Comment:
Same here, if you need a comment, your code could be improved. You could extract this to a self-explaining private method.